### PR TITLE
Support carrier services API

### PR DESCRIPTION
**Pull request by mash**
Support the carrier services API https://shopify.dev/docs/admin-api/rest/reference/shipping-and-fulfillment/carrierservice

### CODE REVIEW

**Review by ysurtayev-bold — COMMENTED** [review]
Please could you add unit tests for your changes?

**Comment by mash** [comment]
Back to coverage 100%! @ysurtayev-bold

**Review by oliver006 — DISMISSED** [review]

**Comment by mash** [comment]
Rebased and force pushed to fix conflict.

File: carrier.go [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+	client *Client
+}
+
+// CarrierResource represents a Shopify carrier service
+type CarrierResource struct {
+	// Whether this carrier service is active.
+	Active bool `json:"active,omitempty"`
+
+	// The URL endpoint that Shopify needs to retrieve shipping rates. This must be a public URL.
+	CallbackURL string `json:"callback_url,omitempty"`
```
[the remark this answers is not in the document]
Comment (mash) [reply carrier.go]: Done!

**Comment by mash** [comment]
and all done!

File: carrier.go [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+}
+
+// CarrierServiceOp handles communication with the product related methods of
+// the Shopify API.
+type CarrierServiceOp struct {
+	client *Client
+}
+
+// CarrierResource represents a Shopify carrier service
+type CarrierResource struct {
```
Comment (oliver006) [inline carrier.go:30]: My bad, I forgot about this one. I just looked at the shopify.dev docs again to see how they name things, can you rename this one to `CarrierService`?

**Review by oliver006 — COMMENTED** [review]
Sorry, one more thing. And then this is good to go.

File: carrier.go (hunk 2 above)
Comment (mash) [reply carrier.go]: Updated!

Comment (oliver006) [reply carrier.go]: Thank you!

**Review by oliver006 — APPROVED** [review]
🥳

**Pull request merged**